File: app/api/audit.py
from typing import Annotated

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.auth.permission import require_permission
from app.database.database import get_db
from app.models.user import User
from app.services.audit_service import list_audit_logs, serialize_audit_log


router = APIRouter(tags=["audit"])


@router.get("/audit-logs")
def get_audit_logs(
    current_user: Annotated[User, Depends(require_permission("audit_view"))],
    database: Annotated[Session, Depends(get_db)],
    page: int = Query(default=1, ge=1),
    page_size: int = Query(default=20, ge=1, le=100),
    action: str | None = None,
    user_id: int | None = None,
    result: str | None = Query(default=None, pattern="^(success|failed)$"),
):
    items, total = list_audit_logs(
        database,
        page=page,
        page_size=page_size,
        action=action,
        user_id=user_id,
        result=result,
    )
    return {
        "items": [serialize_audit_log(item) for item in items],
        "page": page,
        "page_size": page_size,
        "total": total,
    }

File: frontend/src/api/audit.ts
import { http } from './http'

export type AuditResult = 'success' | 'failed'

export interface AuditLogItem {
  id: number
  user_id: number | null
  action: string
  resource_type: string
  resource_id: string | null
  resource_name: string | null
  result: AuditResult
  detail: string | null
  ip_address: string | null
  created_time: string
}

export interface AuditLogResponse {
  items: AuditLogItem[]
  page: number
  page_size: number
  total: number
}

export async function listAuditLogs(params: { page: number; page_size: number; action?: string; user_id?: number; result?: AuditResult }): Promise<AuditLogResponse> {
  const { data } = await http.get<AuditLogResponse>('/audit-logs', { params })
  return data
}
